### PR TITLE
Implement load and flush

### PR DESCRIPTION
**Pull request by yhmo**
Signed-off-by: yhmo <yihua.mo@zilliz.com>

1. sync load logic
2. sync flush logic
3. avoid compiler warning
4. implement getMetrics/getPersistentSegmentInfo/getQuerySegmentInfo
5. add unittest cases
6. improve usability of some param classes
7. add toString() method for each param class
8. refine error message and exception handling
9. use lambok to simplify code
10. add response wrapper for usability
11. fix search error
12. setup CI

### CODE REVIEW

File: src/main/java/io/milvus/client/AbstractMilvusGrpcClient.java [hunk 1]
```diff
@@ -417,704 +164,1389 @@ private FieldData genFieldData(String fieldName, DataType dataType, List<?> obje
...
+        if (partitionNames == null || partitionNames.isEmpty()) {
+            ShowCollectionsRequest showCollectionRequest = ShowCollectionsRequest.newBuilder()
+                    .addCollectionNames(collectionName)
+                    .setType(ShowType.InMemory)
+                    .build();
+
+            // Use showCollection() to check loading percentages of the collection.
+            // If the inMemory percentage is 100, that means the collection has finished loading.
+            // Otherwise, this thread will sleep a small interval and check again.
+            while (true) {
```
Comment (xiaofan-luan) [inline src/main/java/io/milvus/client/AbstractMilvusGrpcClient.java:180]: what about timeout? do we need a timeout for waiting to load?

File: src/main/java/io/milvus/client/AbstractMilvusGrpcClient.java [hunk 2]
```diff
@@ -417,704 +164,1389 @@ private FieldData genFieldData(String fieldName, DataType dataType, List<?> obje
...
 
-        Status response;
-        try {
-            response = blockingStub().createCollection(createCollectionRequest);
+            Status response = blockingStub().createCollection(createCollectionRequest);
 
             if (response.getErrorCode() == ErrorCode.Success) {
-                logInfo("Created collection " + requestParam.getCollectionName() + " successfully!\n{}",
-                        requestParam.toString());
+                logInfo("CreateCollectionRequest successfully!");
```
Comment (xiaofan-luan) [inline src/main/java/io/milvus/client/AbstractMilvusGrpcClient.java]: need at least the collection name here? same as the other ddl requests.

File: src/main/java/io/milvus/client/AbstractMilvusGrpcClient.java (hunk 1 above)
Comment (xiaofan-luan) [reply src/main/java/io/milvus/client/AbstractMilvusGrpcClient.java:180]: same as flush, we should have a time limit of blocking wait, for example 5 mins.

Comment (yhmo) [reply src/main/java/io/milvus/client/AbstractMilvusGrpcClient.java:180]: Sounds good. I will add a parameter to control the time out.

File: src/main/java/io/milvus/client/AbstractMilvusGrpcClient.java (hunk 2 above)
Comment (yhmo) [reply src/main/java/io/milvus/client/AbstractMilvusGrpcClient.java]: In each api method I add a line "logInfo(requestParam.toString());", to print out the input prameters, including the collection name.

Comment (yhmo) [reply src/main/java/io/milvus/client/AbstractMilvusGrpcClient.java]: done

**Review by xiaofan-luan — APPROVED** [review]
lgtm

**Pull request merged**